app/src/layouts/kanban/index.ts

```diff
@@ -0,0 +1,596 @@
+
+					return relation.related_collection as string;
+				}
+
+				return null;
+			});
+
+			const {
+				fields: groupsCollectionFields,
+				sortField: groupsSortField,
+				primaryKeyField: groupsPrimaryKeyField,
+			} = useCollection(groupsCollection);
+
+			const sort = computed(() => {
+				if (groupsSortField.value) return [groupsSortField.value];
+				if (groupsPrimaryKeyField.value?.field) return [groupsPrimaryKeyField.value.field];
+				return [];
+			});
+
+			const groupFieldsToLoad = computed(() => {
+				if (primaryKeyField.value === null || groupTitle.value === null) return [];
+				return [primaryKeyField.value?.field, groupTitle.value];
+			});
+
+			const groupTitleFields = computed(() => {
+				if (isRelational.value) {
+					return groupsCollectionFields.value.filter((field) => field.type === 'string' || field.type === 'text');
+				}
+
+				return null;
+			});
+
+			const {
+				items: relationalGroupsItems,
+				loading: groupsLoading,
+				error: groupsError,
+				changeManualSort: groupsChangeManualSort,
+				getItems: getGroups,
+			} = useItems(groupsCollection, {
+				sort,
+				limit: ref(100),
+				page: ref(1),
+				fields: groupFieldsToLoad,
+				filter: ref({}),
+				search: ref(null),
+			});
+
+			const groups = computed(() => {
+				if (isRelational.value) return relationalGroupsItems.value;
+				return selectedGroup.value?.meta?.options?.choices as Record<string, any>[];
+			});
+
+			return {
+				groups,
+				groupsLoading,
+				groupsError,
+				groupsChangeManualSort,
+				info,
+				fields,
+				groupTitleFields,
+				groupsPrimaryKeyField,
+				groupsSortField,
+				groupsCollection,
+				addGroup,
+				editGroup,
+				deleteGroup,
+				changeGroupTitle,
+				changeGroupSort,
+				isRelational,
+			};
+
+			async function deleteGroup(id: string | number) {
+				const pkField = primaryKeyField.value?.field;
+
+				if (pkField === undefined) return;
+
+				items.value = items.value.filter((item) => item[pkField] !== id);
+
+				await api.delete(`/items/${groupsCollection.value}/${id}`);
+
+				await getGroups();
+			}
+
+			async function addGroup(title: string) {
+				if (groupTitle.value === null) return;
+
+				await api.post(`/items/${groupsCollection.value}`, {
+					[groupTitle.value]: title,
+				});
+
+				await getGroups();
+			}
+
+			async function editGroup(id: string | number, title: string) {
+				if (groupTitle.value === null) return;
+
+				await api.patch(`/items/${groupsCollection.value}/${id}`, {
+					[groupTitle.value]: title,
+				});
+
+				await getGroups();
+			}
+
+			async function changeGroupTitle(id: string | number, newTitle: string) {
+				const pkField = primaryKeyField.value?.field;
+
+				if (groupTitle.value === null || pkField === undefined) return;
+
+				const itemIndex = items.value.findIndex((item) => item[pkField] === id);
+
+				const item = items.value[itemIndex];
+
+				if (item) {
+					item[groupTitle.value] = newTitle;
+				}
```

Comment:
At first i thought this was unused code but it looks like it is intentionally editing an item in `items.value` as side-effect 🤔 
```suggestion

				const itemIndex = items.value.findIndex((item) => item[pkField] === id);

				if (itemIndex in items.value) {
					items.value[itemIndex][groupTitle.value] = newTitle;
				}
```